src/lp_sensitivity.jl

```diff
@@ -0,0 +1,322 @@
+"""
+Returns the basis status of all variables of the problem in standard form, c.f. `_std_matrix(model)`.
+"""
+function _std_basis_status(model::Model)::Vector{MOI.BasisStatusCode}
+    con_types = list_of_constraint_types(model)
+    vars = all_variables(model)
+    var_to_idx = Dict{VariableRef, Int64}()
+    for (j, var) in enumerate(vars)
+        var_to_idx[var] = j
+    end
+    basis_status = fill(MOI.BASIC, length(vars))
+
+    for (F, S) in con_types
+        constraints = all_constraints(model, F, S)
+        if F <: VariableRef
+            for constraint in constraints
+                con_obj = constraint_object(constraint)
+                j = var_to_idx[con_obj.func]
+                basis_status[j] = MOI.get(model, MOI.ConstraintBasisStatus(), constraint)
+                if S <: MOI.LessThan && basis_status[j] == MOI.NONBASIC
+                    basis_status[j] = MOI.NONBASIC_AT_UPPER
+                elseif S <: MOI.GreaterThan && basis_status[j] == MOI.NONBASIC
+                    basis_status[j] = MOI.NONBASIC_AT_LOWER
+                end
+            end
+            #issue 1892 makes is hard to know how to handle variable constraints, atm. assume unique variable constraints.
+            continue
+        end
+        for constraint in constraints
+            con_basis_status = MOI.get(model, MOI.ConstraintBasisStatus(), constraint)
+            if S <: MOI.LessThan && con_basis_status == MOI.NONBASIC
+                con_basis_status = MOI.NONBASIC_AT_UPPER
+            elseif S <: MOI.GreaterThan && con_basis_status == MOI.NONBASIC
+                con_basis_status = MOI.NONBASIC_AT_LOWER
+            end
+            push!(basis_status, con_basis_status)
+        end
+    end
+    return basis_status
+end
+
+function lp_rhs_perturbation_range(constraint::ConstraintRef{Model, _MOICON{F, S}}
+                      )::Tuple{Float64, Float64} where {S <: Union{MOI.LessThan, MOI.GreaterThan, MOI.EqualTo}, T, F <: Union{MOI.ScalarAffineFunction{T}, MOI.SingleVariable}}
+    model = owner_model(constraint)
+    if termination_status(model) != MOI.OPTIMAL
+        error("The perturbation range of rhs is not available because the current solution "*
+              "is not optimal.")
+    end
+
+    try
+        con_status = MOI.get(model, MOI.ConstraintBasisStatus(), constraint)
+        # The constraint is inactive.
+        if con_status == MOI.BASIC
+            if S <: MOI.LessThan
+                return value(constraint) - constraint_object(constraint).set.upper, Inf
+            end
+            if S <: MOI.GreaterThan
+                return -Inf, value(constraint) - constraint_object(constraint).set.lower
+            end
+            return 0.0, 0.0
+        end
+    catch e
+        error("The perturbation range of rhs is not available because the solver doesn't "*
+              "support ´ConstraintBasisStatus´.")
+    end
+    # The constraint is active.
+
+    # Optimization possible, only basic columns needed
```

Comment:
Rephrase "optimization possible" as a TODO, e.g., "TODO: This could be made more efficient because actually we only need the basic columns."